### PR TITLE
Fix optin.cplusplus.UninitializedObject issue

### PR DESCRIPTION
**Pull request by larryluogit**
Changelog category:

- Not for changelog (changelog entry is not required)

Changelog entry (a user-readable short description of the changes that goes to CHANGELOG.md):
This PR fixes the CLANG-TIDY Medium severity errors:

- performance-move-const-arg
- optin.cplusplus.UninitializedObject

### CODE REVIEW

**Comment by larryluogit** [comment]
@alesapin How can I resolve "INFO:root:User 'larryluogit' is not trusted" issue?

File: src/Common/ColumnsHashingImpl.h [hunk 1]
```diff
@@ -125,7 +125,7 @@ class FindResultImpl : public FindResultImplBase, public FindResultImplOffsetBas
 
 public:
     FindResultImpl()
-        : FindResultImplBase(false), FindResultImplOffsetBase<need_offset>(0)
+        : FindResultImplBase(false), FindResultImplOffsetBase<need_offset>(0), value(nullptr)
```
Comment (alexey-milovidov) [inline src/Common/ColumnsHashingImpl.h:128]: We intentionally allow uninitialized value here.

File: src/Common/ColumnsHashing.h [hunk 2]
```diff
@@ -503,7 +503,7 @@ struct HashMethodKeysFixed
         return true;
     }
 
-    HashMethodKeysFixed(const ColumnRawPtrs & key_columns, const Sizes & key_sizes_, const HashMethodContextPtr &)
+    HashMethodKeysFixed(const ColumnRawPtrs & key_columns, const Sizes & key_sizes_, const HashMethodContextPtr &) noexcept
```
Comment (alexey-milovidov) [inline src/Common/ColumnsHashing.h:506]: Why noexcept?

File: src/Common/JSONBuilder.h [hunk 3]
```diff
@@ -61,7 +61,7 @@ class JSONNumber : public IItem
 class JSONBool : public IItem
 {
 public:
-    explicit JSONBool(bool value_) : value(std::move(value_)) {}
+    explicit JSONBool(bool value_) : value(value_) {}
```
Comment (alexey-milovidov) [inline src/Common/JSONBuilder.h:64]: Ok.

**Comment by alexey-milovidov** [comment]
You should add

> performance-move-const-arg
optin.cplusplus.UninitializedObject

to `.clang-tidy` file, otherwise the changes are not checked in the CI.

**Comment by alexey-milovidov** [comment]
> How can I resolve "INFO:root:User 'larryluogit' is not trusted" issue?

Manual approve needed.

File: src/Common/ColumnsHashingImpl.h (hunk 1 above)
Comment (larryluogit) [reply src/Common/ColumnsHashingImpl.h]: a comment is added instead.

File: src/Common/ColumnsHashing.h (hunk 2 above)
Comment (larryluogit) [reply src/Common/ColumnsHashing.h]: removed

**Comment by larryluogit** [comment]
Enabled optin.cplusplus.UninitializedObject in .clang-tidy
performance-move-const-arg is an existing check in .clang-tidy

**Comment by larryluogit** [comment]
@alexey-milovidov Could you review the changes again please?

**Comment by alexey-milovidov** [comment]
@mergifyio update

**Review by alexey-milovidov — APPROVED** [review]

**Comment by alexey-milovidov** [comment]
The changes are 100% ok, but I need to wait before all checks will be green.

**Pull request merged**